src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,93 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  ids: string;
+}
+
+class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_RESTORE;
+  }
+
+  public get description(): string {
+    return 'Restores given items from the site recycle bin';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Restoring items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
+
+    const ids: string[] = this.splitIdsList(args.options.ids);
+    const idsChunks: string[][] = [];
+
+    while (ids.length) {
+      idsChunks.push(ids.splice(0, 20));
+    }
+
+    idsChunks.forEach(async (idsChunk: string[], index: number) => {
```

Comment:
We're missing error handling. Since we're using `await` the whole loop should be wrapped a `try..catch`, which isn't that great. What if we'd change it to use `map` instead of `forEach` so that we get an array of Promises back and then using `Promise.all` we can wait until they all execute and call `cb()` after they're resolved and handle errors when they're rejected?

Reply:
done